app/experimenter/experiments/serializers.py

```diff
@@ -616,7 +616,36 @@ def update(self, instance, validated_data):
         return instance
 
 
-class ExperimentDesignPrefSerializer(ExperimentDesignBaseSerializer):
+class PrefValidationMixin(serializers.ModelSerializer):
+
+    def validate_pref(self, pref_type, pref_value, error_list):
+        if pref_type == "integer":
+            try:
+                int(pref_value)
+            except ValueError:
+                error_list.append({"value": "The pref value must be an integer."})
```

Comment:
Isn't `"value"` here specific to the field on the given model? This should probably be a parameter.